### DISCUSSION TITLE
Text Editors ou IDEs - Qual estão usando?

### DISCUSSION THREAD

**Discussion by felipefialho** [frontendbr/forum#42] [Perguntas e Respostas]
Lançada a proza com o @fdaciuk no [Twitter](https://twitter.com/fdaciuk/status/698104267115261952). Vou trazer a questão aqui. 

Comentei com ele que gosto tanto do Sublime Text 2, que não consigo desapegar. Acredito que mesmo após tanto tempo ele continua funcionando perfeitamente.

O @diegoeis escreveu um post no Tableless contando que o [Sublime Text está sendo desenvolvido novamente](http://tableless.com.br/sublime-text-esta-sendo-desenvolvido-de-novo/). Vai vir novidades por ai.

Vou aproveitar para trazer o mago do VIM, @matmarsiglio na discussão.

E aí, qual ou quais, vocês estão utilizando?

**Comment by doug2k1** [comment]
Usava muito o Sublime, mas devido à alta do dólar e a falta de atualizações, fui experimentar o Atom. Estou gostando muito. Removi um monte de plugins (ele vem com muita coisa de fábrica), instalei outros, e hoje está me atendendo muito bem.

Quero experimentar o WebStorm, mas ainda não fiz isso.

**Comment by karlprieb** [comment]
Fiquei um bom tempo com o Sublime. Há 1.5 anos eu fui para o Atom, mas nos últimamente voltei para o Sublime. O meu projeto atual tem uns JSON de +/- 3k linhas e o Atom engasga para abrir esses JSON :(

**Comment by vitorbritto** [comment]
Utilizo VIM e Atom atualmente. Como o @karlprieb disse, o Atom engasga quando um arquivo com mais de 2k de linhas é aberto. Entretanto, gosto bastante da UI do Atom e das features nativas dele. É um bom editor, mas não encontrei (até o momento) algum editor melhor que o VIM.

**Comment by fdaciuk** [comment]
Comecei a usar o Vim por necessidade.. Sublime estava me zuando, e o Atom consumindo toda a minha memória.

Agora não quero mais largar o vim :heart: 

_PS: Mas vou testar o sublime novo, para momentos de **extrema** necessidade xD_

**Comment by raphaklaus** [comment]
@LFeh boa issue cara!

Usei o Brackets durante um tempo e achei ele muito bom por já ter uma série de funcionalidades embutidas. Acho ele mais intuitivo que o Sublime e o live preview é muito bacana. Entretanto, o achei muito pesado.

O Sublime Text me acompanha há alguns anos já (na real, desde o lançamento dele, foi amor à primeira vista heheh). É um editor extremamente leve, minimalista no visual, customizável, tem grande número de packages...

Não sou muito fã de IDE's, mas já usei várias (RubyMine, WebStorm, etc). Sempre muito boas e com funcionalidades ao alcance de um atalho, porém, devido ao grande número de features ativas, o editor fica muito pesado. Esse foi o motivo pelo qual parei com IDE's faz anos.

**Comment by baltazarparra** [comment]
@fdaciuk, @vitorbritto como vocês utilizam? Poderiam postar um fluxo de trabalho, tipo, plugins, extenções e tals para alguém que nunca trabalhou com ele.

**Comment by ghost** [comment]
@ravenNNN  eu não utilizo, mas tem um screencast do Akita bem maneiro.. [Aqui o link :)](http://www.akitaonrails.com/2010/07/19/screencast-comecando-com-vim)

**Comment by ghost** [comment]
Eu utilizo Sublime Text, já decorei os packages, mas meu objetivo é até 2025 estar utilizando VIM.. se estiver disponível até lá xD 

Mas falando sério, com 1 ano de uso conseguimos ser produtivos com VIM? :wq ops :wq! heuheu

**Comment by fdaciuk** [comment]
@ravenNNN, hoje eu estou usando essas configs aqui: https://gist.github.com/fdaciuk/061fa54ebf66d9f96a10

Mas se você nunca mexeu com Vim, recomendo antes usar o `vimtutor` (só rodar no seu terminal esse comando, se tiver o vim instalado), que ele vai te dar as instruções iniciais.

A curva de aprendizado é meio alta.. eu já uso a um tempo para fazer pequenas edições, depois comecei a usar para alguns projetos open source, fora do horário de trabalho, pra ir acostumando com ele. 

Depois que eu consegui deixar ele com essas configurações do link acima, ae não larguei mais xD

**Comment by baltazarparra** [comment]
Eu já uso ele pra pequenas edições, e realmente queria adotar como minha ide, não conhecia esse vimtutor @fdaciuk Obrigado.

**Comment by mtmr0x** [comment]
O problema de argumentar sobre o que usa, é que tem que fazer sentido pra vc. 

Eu quando comecei a usar o vim, foi quase uma necessidade se encontrando com uma vontade de aprender algo novo. A curva de aprendizado de vim é infinita (sem exageros, a toda hora vc descobre algo novo), mas a curva de aprendizado da lógica de uso do vim é bem tranquila.

Eu recomendo o vim para todo mundo, mas sempre depende muito da aceitação da pessoa. Normalmente as pessoas não querem aceitar que o vim pode ajudar e aí rola os comentários negativos haha.

@ravenNNN, tem um post do @woliveiras que da uma repassada nuns topicos importantes do vimtutor: http://woliveiras.com.br/vim-para-noobs/livro/chapters/inicio.html

**Comment by vitorbritto** [comment]
@ravenNNN tenho um [setup](https://github.com/vitorbritto/setup-vim) aqui no GitHub. Existem projetos que facilitam a vida dos novatos que desejam utilizar o VIM, como é o caso do [spf13](http://vim.spf13.com/).

Você pode [aprender VIM de forma divertida](http://vim-adventures.com/) e com [casts interessantes também](http://vimcasts.org/).

**Comment by DevCarlosAugusto** [comment]
IDE eu uso o Visual Studio no trabalho oficial, melhor IDE que já usei, completa em tudo, porém muito pesada, poderia ter uma área de customização para cortar coisas que não fossem usadas.

Mas em casa e em projetos particulares/menores eu uso o Atom/Brackets atualmente e o Sublime pra abrir arquivos muito grandes, como o pessoal disse, eles costumam travar com arquivos muito grandes, principalmente JSON's, espero por melhorias deles ou do sublime, acho que o sublime podia aproveitar pra ter coisas que o Atom/Brackets trouxeram e são muito boas.....

**Comment by tcelestino** [comment]
Para quem usava o FrontPage, Dreamweaver (para codar e não arrastar), usar hoje o Sublime ou o Atom é um lucro enorme. 

Brincadeira a parte, tem um tempo que sou usuário do Sublime e tem algumas semanas (desde do último meetupCSS) que estou usando o Atom. Estou gostando, apesar do mesmo engasgar em diversas situações.

**Comment by woliveiras** [comment]
Estou usando o Sublime em conjunto com o Vim.

Comecei a usar o Vim por que um monte de gente falava muito bem dele e eu achava que era só para eles se sentirem fodões que usavam esse editor, pois, das vezes que usei, não me senti tão produtivo. - Por isso a zoeira no meu post falando que o cara que usa Vim vai se sentir mais "hackudo". ;D

Acabou que comecei a gostar tanto de não precisar sair do Terminal ou de tirar a mão do teclado que estou incluindo o Vim aos poucos no dia a dia. Estou gostando muito. Só fica um pouco ruim o vício de não usar tanto o mouse, pois nem tudo o que usamos no dia a dia tem tanto atalho maneiro! kkkk

Adicionei no meu `.vimrc` alguns plugins que eu falo [nesse post](http://woliveiras.com.br/vim-para-noobs/livro/chapters/alguns-plugins-maneiros.html).

Como o @matmarsiglio falou, todos os dias eu aprendo algo novo no Vim! É bem louco.

**Comment by kazzkiq** [comment]
Migrei do **Sublime** pro **Atom** no meio do ano passado. Até agora, nada a reclamar... **a não ser a performance**.

O Sublime é imbatível nesse quesito, tanto em inicialização quanto em número de janelas abertas. Se eu abro 2 workspaces no Atom ele já começa a dar umas engasgadas. No Sublime você abre 20 janelas e sua CPU ta lá de boa em 2%. :laughing: 

Mas pra mim o Atom ainda dá umas regalias _out-of-the-box_ que fazem dele a minha escolha.

**Comment by marinhoOliv** [comment]
Galera, sou fã demais do sublime, e por necessidade usei o https://c9.io/ e curti. Fica a dica ;)

**Comment by rickbenetti** [comment]
Comecei com PageMill da Adobe em 1996, depois migrei pro Notepad até 2003, aí descobri o Notepad ++ fiquei nele até o Sublime Text 2, Sublime Text  3 (comprei a licença antes do lançamento da versão 3), agora estou no Atom, já usei Aptana, PhpStorm, mas acho essas IDEs muito pesadas, VIM e VI eu já tentei, não gosto de interface puro teclado pois esqueço muito rápido os comandos, preciso do mouse pra algumas tarefas, não dá pra confiar nos meus dedos vivo fechando janelas por engano direto.

**Comment by yuricamara** [comment]
Webstorm pra projetos. Testei quase todos editores e cheguei a conclusão que é o melhor pra mim. Gosto de tudo integrado num lugar só. E ele faz isto bem. Pra só abrir arquivo aí vou de Sublime ou Visual Studio Code.

**Comment by yuricamara** [comment]
Quase todos editores famosos.

**Comment by caiorss** [comment]
Emacs. Apesar de ser uma máquina Lisp de mais de 35 anos de desenvolvimento, pode se tornar um excelente editor quando bem configurado. Sem sombra de duvidas é um dos melhores editores de texto que já usei. O que diferencia o Emacs é que ele é totalmente programável, configurável e integrável a ferramentas externas. No Emacs sempre é possível criar novas funcionalidades e comandos usando sua linguagem de programação própria Elisp que é parecido com o Common LIsp. O Emacs também é útil para criar documentação com o org-mode que pode ser exportado para o Mardown do github, html, latex e pdf. Outro editor do qual gostei é o Geany que é leve e rápido e com suporte a milhares de linguagens.

Eu criei um tutorial de como programar e configura o Emacs, o qual se encontra disponível em: [Emacs in a Box](https://github.com/caiorss/Emacs-Elisp-Programming)

**Comment by lnfnunes** [comment]
**_Sublime Text_**
É o mais leve pra quem precisa de performance.
O problema que vejo, principalmente quando se fala em comunidade desenvolvedora, é que 90% (dados auditados pelo IBGE) não seguem essa nota abaixando, achando que ele é totalmente free.
_"Sublime Text may be downloaded and evaluated for free, however a license **must be purchased** for continued use."_

**_WebStorm_**
É o mais poderoro, afinal não é um editor, é uma IDE.
Porém seu custo anual talvez seja o maior bloqueante de alavancar a preferência da comunidade.
Já no meio corporativo, essa barreira pode ser quebrada mais facilmente e acredito que tenhamos bons adeptos.

**_Atom_**
Na minha opinião é o meio termo entre os dois. Um editor com cara de IDE! :smile: 

**_Brackets_**
Outro bom editor que vem ganhando espaço e possui uma série de plugins e integrações bem legais.

**_Vim_**
Não sou tão hardcore, então minha utilização nele se resume a editar arquivos rapidamente via terminal. Principalmente configurações do SO, commits de SCM (git ou svn), etc...
Vejo muita gente comentando que a produtividade com ele aumenta absurdo, mas talvez por eu não ter esse conhecimento tão a fundo, não consigo avaliar tanto assim. Acho ele muito prático, sim, nos casos que citei acima.

Atualmente **eu utilizo o Atom** e me atende muito bem.
Praticamente não preciso ir no terminal, executo os comandos direto nele e sem mouse.
Não é tão leve quando o Sublime, porém não chega a causar desconforto pelo peso. Basicamente só engasga com arquivos muito grandes (como já foi comentado https://github.com/frontendbr/forum/issues/42#issuecomment-183326194).
Possui algumas features e configuração em sua instalação inicial que no Sublime, por ser minimalista ao extremo, precisaria instalar alguns plugins.

Enfim, utilize o editor/ide que mais se sentir confortável! Afinal, é nele que vai passar a maior parte do seu dia.
Outra coisa muito importante **entenda tudo sobre ele**, configurtações, teclas de atalho, plugins... isso é oq fará toda a diferença no seu workflow.

**Comment by jhonnymoreira** [comment]
Uso muito o Sublime Text, mas por estar me adaptando rápido e bem ao tmux, to pensando em embarcar no vim. O que faz meu workflow me agradar com Sublime é a facilidade que tenho pra achar os meus arquivos, os comandos que os dedinhos já fluem, além da interface leve e personalizável. Já tentei o Atom, mas não me agradou. =/

**Comment by rmjunior** [comment]
Vish, Eu sou Atom na veia! 
Como você pode personalizar ele com JS, você consegue adaptar ao seu workflow.
Ele tem meus atalhos de teclado integrado com o gulp, pra mim não rodar o Gulp desnecessário (isso pode ser trocado se eu reescrever todo script pra usar gulp changed, e eu tenho que fazer uma lógica do carario pra dar certo na estrutura dos projetos que atuo), criar um server do .net usando mono pra rodar meu projeto, usando o .net 64 e multiprocessador ( da build num projeto numa velocidade fantástica #chupamicrosoftvisualstudio ), criar algumas estruturas a partir de um template de telas padrões daqui da empresa, criar alguns json a partir de atalhos pra gerar disparo de construtores do framework daqui da @happybiz.

Ainda rodo dentro dele uma espécie de Query-Analyzer do SQL Server, ele é muito básico, mais adianta bastante na hora que vou criar minhas procedures e functions, que também uso templates e atalhos que eu criei pra gerar coisas repetitivas. O resto eu tenho que fazer dentro da IDE do SQL Server mesmo.

Uso ele pra conectar também no Aws Redshift e no Aws dynamodb. No mesmo esquema que uso com SQL Server. 

O editor está avançando bastante, tão reduzindo o uso de coffescript nele e tem ficado mais rápido.

Pra ser melhor só se eu conseguisse deixa-lo dentro de um repositório do github, mais o fdp usa algumas pastas no appData e em todas as vezes que eu tentei faltou arquivo pra colocar lá.
Eu ainda vou fazer um shellscript pra fazer isso, to sem tempo agora.

Não vou ser fanático, me falaram muito bem do WebStorm e do Vim.
Tenho muita curiosidade de mudar, mas não vejo problema atualmente no meu editor.
Ainda posso integrar muito mais coisas nele, como até mover minha demanda no acelera.to, criar ambiente de teste para os testers, fazer mais um monte de tarefinha que gasta tempo no dia a dia.
Inclusive eu tava pensando numa forma de mostrar as demandas do dia e já integrar no github com as alterações do arquivo conforme a demanda a ser executada e eu nem sei se daria certo mas seria show demais.

Atom <3

**Comment by victorserpac** [comment]
@rmjunior sensacional velho, eu uso o Atom há uns 2 meses e já vi o poder dele, mas não tinha uma ideia de todo o potencial dele.

**Comment by renatogalvones** [comment]
Aproveitando o tópico, tive uns problemas pra atualizar o **Atom** atrás do **proxy** aqui da empresa.

Achei a solução nas issues do Atom e tô compartilhando um gist caso alguém também precise.

https://gist.github.com/renatogalvones/e54147330a2793a196b1e47e426e9c64

**Comment by barbier** [comment]
Já usei o Sublime, mas perdi o costume porque ele é pago e 60 Obamas não estão na minha realidade atual;
Usei Notepad ++ (somente Windows) e ele é bem customizável, usei para fazer algumas coisas "divertidas"....
Fiquei bastante tempo no Brackets porque ele é leve, mas nem com plugins ele fica bom pra trabalhar com linguagens de back-end;
Atualmente estou com o Atom, mas ele trava aleatoriamente e estou começando a me estressar com isso...

**Comment by fabricionaweb** [comment]
@barbier o Sublime é **trial sem limites**, é só apertar esc as vezes... melhor isso do que esses problemas de travamento citado e uso de memória rsrs

**Comment by rafaelverger** [comment]
@fabricionaweb o Sublime **não** é _trial sem limites_... Ele pode ser baixado e testado sem custos, porém, para o uso contínuo do mesmo, os desenvolvedores esperam que você compre uma licença;

![image](https://cloud.githubusercontent.com/assets/588697/16093467/6227a152-3312-11e6-957f-5bc5845dace3.png)

**Comment by fabricionaweb** [comment]
Talvez seja apenas uma questão de interpretação e consciência ☺️ Nada que impeça seu uso.

![screen shot 2016-06-15 at 16 10 37](https://cloud.githubusercontent.com/assets/15933/16093815/d1682478-3313-11e6-8197-5a4d604ad920.png)

**Comment by rafaelverger** [comment]
Não é interpretação; **must** determina obrigação, o fato dele não bloquear o uso não isenta o usuário de aderir a obrigatoriedade, e então, comprar a licença.

Mas claro, você pode usar ele o quanto você quiser (sabendo que você está errado) sem pagar.

**Comment by felipefialho** [comment]
Indiferente a essa questão de comprar ou não a licença.

Acham que compensa uma mudança do Sublime Text 2 para o 3?

**Comment by edoura** [comment]
Estou no Sublime tbm, até o final do ano rola uma licença.
Agora, qual é a posição da moçada aqui em relação ao [c9](https://c9.io/)? vale a pena? é uma boa mesmo manter as workspace lá, com a equipe? fica o histórico de cada linha específica escrita pelo dev no git? Tenho pensado nisso, e achado maneiro a possibilidade de não precisar do seu pc propriamente. Ta em algum lugar tem internet, só logar e ter seus projetos lá. Agora, é meio palha se não tiver internet. 
Só tenho observado, nada na prática. O que vcs acham?

**Comment by barbier** [comment]
@LFeh eu sinceramente não vi diferença, mas é possível que eu não tenha usado algo complexo o bastante para notar diferença....

**Comment by felipefialho** [comment]
Pelo sim e pelo não, atualizei :)

**Comment by diessica** [comment]
@LFeh,
a performance do Sublime Text 3 é superior, tanto na inicialização quanto nas ações em geral (_Find and Replace_, por exemplo, é mais rápido na versão 3). Também existem novas funcionalidades, como a atualização automática. 

A maioria dos plugins são desenvolvidos já para ST3. Eu não vejo razão alguma para continuar na versão 2.

**Comment by filipelinhares** [comment]
Atualmente uso o [vim](https://github.com/filipelinhares/dotvim) e se precisar de um replace global com uma interface mais bonitinha eu abro no Atom.

**Comment by heliomsolivas** [comment]
Eu sempre usei o Sublime mas onde eu trabalho pagam o **WebStorm** e eu acho ele completão, tem tudo, só trava minha máquina as vezes com o combo "varias abas do chrome + 2 janelas do webstorm".

**Comment by felipefialho** [comment]
Pode crê @diessica. Percebi essa melhora importante na performance no _Find and Replace_. Estou gostando bastante.

Não tinha atualizado ainda por preguiça mesmo, haha.

**Comment by fdaciuk** [comment]
Só o `grep` é mais rápido que o `find` do sublime :stuck_out_tongue_winking_eye:

**Comment by gabrielmaximo** [comment]
galera, hoje, o que diriam da Evolução do Atom?? e o VScode? to na duvida de qual Editor de Texto usar, atualmente estou usando Atom, esta me atendendo sem nenhum travamento, mas  tenho medo de quando eu realmente precisar de poder, ele travar. alguém aqui ainda pode me ajudar?

**Comment by rictorres** [comment]
migrei do sublime3 pro vscode há uma semana e to muito feliz. acho que o vscode engoliu o atom em termos de comunidade e features :)

**Comment by brunover** [comment]
O vscode está recebendo atualizações quase mensalmente e melhorou muito do
começo de 2017 pra cá. Virou minha IDE principal. Somente pra Java ainda
uso o eclipse.

Em 12 de fev de 2018 11:06 AM, "Ricardo Torres" <notifications@github.com>
escreveu:

**Comment by tarciozemel** [comment]
Vim, Neovim, VSCode ou PhpStorm, conforme a necessidade/humor do dia.

**Comment by gabrielmaximo** [comment]
saquei, acho que vou tentar me acostumar com o VS Code, obrigado 👍

**Comment by danielschmitz** [comment]
Uso vscode pra tudo. Ótimo editor e com muitos pluguins. A integração  com
git e o terminal são diferenciais em relação ao sublime.

Atom sempre trava comigo ao editar meus livros com arquivos md acima de
1000 linhas

Em 12 de fev de 2018 15:26, "Gabriel Maximo" <notifications@github.com>
escreveu:
